api/src/main/java/org/pmiops/workbench/compliance/ComplianceServiceImpl.java

```diff
@@ -21,18 +23,35 @@
   private static final String MOODLE_EXCEPTION = "moodle_exception";
   private static final String MOODLE_USER_NOT_ALLOWED_ERROR_CODE = "guestsarenotallowed";
   private CloudStorageService cloudStorageService;
+  private Provider<WorkbenchConfig> configProvider;
+
 
   @Autowired
-  public ComplianceServiceImpl(CloudStorageService cloudStorageService) {
+  public ComplianceServiceImpl(CloudStorageService cloudStorageService,
+      Provider<WorkbenchConfig> configProvider) {
     this.cloudStorageService = cloudStorageService;
+    this.configProvider = configProvider;
   }
 
   private String getToken() {
     return this.cloudStorageService.getMoodleApiKey();
   }
 
+  private boolean enableMoodleCalls() {
+    return configProvider.get().moodle.enableMoodleBackend;
+  }
+
+  private void setApiHost() {
+    ApiClient apiClient = api.getApiClient();
+    apiClient.setBasePath("https://" + configProvider.get().moodle.host +"/webservice/rest");
+  }
+
   @Override
   public Integer getMoodleId(String email) throws ApiException {
+    if (!enableMoodleCalls())
+      return null;
+    setApiHost();
```

Comment:
Manually setting this before every call seems like a lot of duplication. Perhaps it would make more sense to copy the approach we use for the other API clients: add a `MoodleConfig.java` file in this directory similar to https://github.com/all-of-us/workbench/blob/dd11c400fade01078bfd32571a175ceae08d9ffd/api/src/main/java/org/pmiops/workbench/firecloud/FireCloudConfig.java

and provide a properly configured `MoodleApi` from there. Then this service takes a `Provider<MoodleApi>` and uses that for calls.